src/main/java/com/thealgorithms/sorts/SelectionSortRecursive.java

```diff
@@ -0,0 +1,66 @@
+package com.thealgorithms.sorts;
+
+/**
+ * Class that implements the Selection Sort algorithm using recursion.
+ */
+public class SelectionSortRecursive implements SortAlgorithm {
+
+    /**
+     * Sorts an array using recursive selection sort.
+     *
+     * @param array the array to be sorted
+     * @param <T>   the type of elements in the array (must be Comparable)
+     * @return the sorted array
+     */
+    public <T extends Comparable<T>> T[] sort(T[] array) {
+        recursiveSelectionSort(array, 0);
+        return array;
+    }
+
+    /**
+     * Recursively sorts the array using selection sort.
+     *
+     * @param array the array to be sorted
+     * @param index the current index to start sorting from
+     * @param <T>   the type of elements in the array (must be Comparable)
+     */
+    private static <T extends Comparable<T>> void recursiveSelectionSort(T[] array, int index) {
+        if (array == null || array.length == 0) {
```

Comment:
```suggestion
        if (array.length == 0) {
```

By the way: the `null` check _should_ happen at the level of `SortAlgorithm`.

Reply:
Thanks. Yes, in other places this is on level before. Moved there.